### PR TITLE
msal-react FAQ and class component doc

### LINKED ISSUE #2933: Does MSAL-React support class components?
## Library

- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [ ] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-node@1.x.x`
- [X] `@azure/msal-react@1.0.0-alpha.3`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angular@2.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description

Am using https://www.npmjs.com/package/@azure/msal-react in my react project. The library provides hooks to perform auth very easliy.

So in functional component, for getting access token,
```
if (account && inProgress === "none") {
instance.acquireTokenSilent({
...loginRequest,
account: account
}).then((response) => {
callMsGraph(response.accessToken).then(response => setGraphData(response));
});
}
```
needs to be called where ```const { instance, accounts, inProgress } = useMsal();``` is used.

But I need to call api to fetch data from class components.
So, how to achieve the same functionality in class component
## Source

- [X] Internal (Microsoft)
- [ ] Customer request

### PR DESCRIPTION
**Pull request by tnorling**
Adds an initial FAQ for `msal-react` + a doc for use with class components. Closes #2933

### CODE REVIEW

**Review by jasonnutter — APPROVED** [review]

File: lib/msal-react/docs/class-components.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
+# Using msal-react with class components
+
+`msal-react` does support both function components and class components. However, you will not be able to use `msal-react` hooks inside your class components so if you need access to authentication state inside your class component you will need to use `msal-browser` directly to obtain similar functionality.
+
+## Initialization
+
+Just like when using function components you will need an `MsalProvider` component at the top level of the component tree that requires access to authentication state.
```
Comment (jo-arroyo) [inline lib/msal-react/docs/class-components.md:7]:
```suggestion
Just like when using function components you will need a `MsalProvider` component at the top level of the component tree that requires access to authentication state.
```

**Review by jo-arroyo — APPROVED** [review]
A few nits, otherwise LGTM!

File: lib/msal-react/FAQ.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# FAQ
+
+***
+**[Compatibility](#compatibility)**
+
+1. [What versions of React are supported](#What-versions-of-React-are-supported)
```
Comment (jo-arroyo) [inline lib/msal-react/FAQ.md:6]:
```suggestion
1. [What versions of React are supported](#what-versions-of-React-are-supported)
```
Current links don't seem to work

File: lib/msal-react/FAQ.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+1. [Does msal-react support Server Side Rendering (SSR) or static site generation](#Does-msal-react-support-Server-Side-Rendering-SSR-or-static-site-generation)
```
Comment (jo-arroyo) [inline lib/msal-react/FAQ.md:7]:
```suggestion
1. [Does msal-react support Server Side Rendering (SSR) or static site generation](#does-msal-react-support-Server-Side-Rendering-SSR-or-static-site-generation)
```

File: lib/msal-react/FAQ.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+1. [Does msal-react support class components](#Does-msal-react-support-class-components)
```
Comment (jo-arroyo) [inline lib/msal-react/FAQ.md:8]:
```suggestion
1. [Does msal-react support class components](#does-msal-react-support-class-components)
```

File: lib/msal-react/docs/class-components.md (hunk 1 above)
Comment (tnorling) [reply lib/msal-react/docs/class-components.md:7]: I don't think this is right, would love to be educated though haha

Comment (jo-arroyo) [reply lib/msal-react/docs/class-components.md:7]: Oh, I think you're right, actually. I forgot to read it out loud. Sorry!

**Pull request merged**